File: 0217-Contains-Duplicate/0217.java
import java.util.Arrays;
import java.util.HashMap;
import java.util.HashSet;
import java.util.Map;
import java.util.Set;

class Solution {
    public boolean containsDuplicate(int[] nums) {
        Map<Integer, Integer> hash = new HashMap<>();

        for (int num : nums) {
            if (hash.containsKey(num))
                return true;
            hash.put(num, 1);
        }

        return false;
    }

    public boolean containsDuplicate1(int[] nums) {
        Set<Integer> set = new HashSet<>(Arrays.stream(nums).boxed().toList());
        return nums.length != set.size();
    }
}

File: 0217-Contains-Duplicate/0217.py
from typing import List


class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        hash = {}

        for num in nums:
            if num in hash.keys():
                return True

            hash[num] = 1
        
        return False
        
    def containsDuplicate1(self, nums: List[int]) -> bool:
        return len(nums) != len(set(nums))